src/SignatureHelpProvider.php

```diff
@@ -0,0 +1,98 @@
+<?php
+declare(strict_types = 1);
+
+namespace LanguageServer;
+
+use Microsoft\PhpParser\Node\DelimitedList\ArgumentExpressionList;
+use Microsoft\PhpParser\Node\Expression\CallExpression;
+use Microsoft\PhpParser\Node\Expression\ArgumentExpression;
+use LanguageServer\Index\ReadableIndex;
+use LanguageServer\Protocol\{
+    Range,
+    Position,
+    SignatureHelp,
+    SignatureInformation,
+    ParameterInformation
+};
+
+class SignatureHelpProvider
+{
+    /**
+     * @var DefinitionResolver
+     */
+    private $definitionResolver;
+
+    /**
+     * @var ReadableIndex
+     */
+    private $index;
+
+    /**
+     * @param DefinitionResolver $definitionResolver
+     * @param ReadableIndex      $index
+     */
+    public function __construct(DefinitionResolver $definitionResolver, ReadableIndex $index)
+    {
+        $this->definitionResolver = $definitionResolver;
+        $this->index = $index;
+    }
+
+    /**
+     * Returns signature help for a specific cursor position in a document
+     *
+     * @param PhpDocument $doc The opened document
+     * @param Position $pos The cursor position
+     * @return SignatureHelp
+     */
+    public function provideSignature(PhpDocument $doc, Position $pos) : SignatureHelp
+    {
+        $node = $doc->getNodeAtPosition($pos);
+        $arge = null;
+        while ($node &&
+            !($node instanceof ArgumentExpressionList) &&
+            !($node instanceof CallExpression) &&
+            $node->parent
+        ) {
+            if ($node instanceof ArgumentExpression) {
+                $arge = $node;
+            }
+            $node = $node->parent;
+        }
+        if (!($node instanceof ArgumentExpressionList) &&
+            !($node instanceof CallExpression)
+        ) {
+            return new SignatureHelp;
```

Comment:
There is no ```label``` property in the ```SignatureHelp``` object - maybe I am missin something - please explain? This is supposed to be an empty result (no nested ```SignatureHelpInformation``` objects).